cyvlfeat/fisher/cyfisher.pyx

```diff
@@ -0,0 +1,67 @@
+# distutils: language = c
+# Copyright (C) 2007-12 Andrea Vedaldi and Brian Fulkerson.
+# All rights reserved.
+
+# This file is modified from part of the VLFeat library and is made available
+# under the terms of the BSD license.
+import numpy as np
+cimport numpy as np
+cimport cython
+
+# Import the header files
+from cyvlfeat._vl.fisher cimport vl_fisher_encode
+
+@cython.boundscheck(False)
+cpdef cy_fisher(np.ndarray[float, ndim=2, mode='c'] X,
+                np.ndarray[float, ndim=2, mode='c'] MEANS,
+                np.ndarray[float, ndim=2, mode='c'] COVARIANCES,
+                np.ndarray[float, ndim=1, mode='c'] PRIORS,
+                bint Normalized,
+                bint SquareRoot,
+                bint Improved,
+                bint Fast,
+                bint Verbose):
+
+    vl_size numClusters = MEANS.shape[1];
+    vl_size dimension = MEANS.shape[0];
+    vl_size numData = X.shape[1];
+    int flags = 0;
+
+    if Normalized:
+        flags |= VL_FISHER_FLAG_NORMALIZED
+    
+    if SquareRoot:
+        flags |= VL_FISHER_FLAG_SQUARE_ROOT
+        
+    if Improved:
+        flags |= VL_FISHER_FLAG_IMPROVED
+        
+    if Fast:
+        flags |= VL_FISHER_FLAG_FAST
+    
+    if Verbose:
+        print 'vl_fisher: num data: %d' % numData
+        print 'vl_fisher: num clusters: %d' % numClusters
+        print 'vl_fisher: data dimension: %d' % dimension
+        print 'vl_fisher: code dimension: %d' % 2 * numClusters * dimension
+        print 'vl_fisher: square root: %d' % SquareRoot
+        print 'vl_fisher: normalized: %d' % Normalized
+        print 'vl_fisher: fast: %d' % Fast
+
+    enc = np.zeros((2*numClusters*dimension,),dtype=np.float32)
```

Comment:
You can type this variable e.g. `cdef float[:] enc = np.zeros((2*numClusters*dimension,),dtype=np.float32)`